Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs

```diff
@@ -83,72 +85,48 @@ private bool GetVisibleSetting(string settingName)
-            {
-                return false;
-            }
-
-            var version = DotNetNukeContext.Current.Application.Version;
-            var request = HttpContext.Current.Request;
-
-            var imageUrl = Upgrade.UpgradeIndicator(version, request.IsLocal, request.IsSecureConnection);
-            imageUrl = Globals.AddHTTP(imageUrl.TrimStart('/'));
-
-            try
-            {
-                string hash;
-                const string cacheKey = "UpdateServiceUrlCacheKey";
-                var cachedData = DataCache.GetCache(cacheKey) as string;
-                if (cachedData != null)
-                {
-                    hash = cachedData;
-                }
-                else
-                {
-                    var webRequest = WebRequest.CreateHttp(imageUrl);
-                    webRequest.Timeout = Host.WebRequestTimeout;
-                    webRequest.UserAgent = request.UserAgent;
-                    webRequest.Referer = request.RawUrl;
-
-                    using (var stream = ((HttpWebResponse)webRequest.GetResponse()).GetResponseStream())
-                    {
-                        if (stream == null)
-                        {
-                            return false;
-                        }
-
-                        using (var sha256 = SHA256.Create())
-                        {
-                            hash =
-                                BitConverter.ToString(sha256.ComputeHash(stream)).Replace("-", string.Empty).ToLowerInvariant();
-                            DataCache.SetCache(cacheKey, hash, (DNNCacheDependency)null,
-                                Cache.NoAbsoluteExpiration, TimeSpan.FromDays(1), CacheItemPriority.Normal, null);
-                        }
-                    }
-                }
-
-                switch (hash)
-                {
-                    case NormalUpdateHash:
-                        updateType = UpdateType.Normal;
-                        return true;
-                    case CriticalUpdateHash:
-                        updateType = UpdateType.Critical;
-                        return true;
-                    default:
-                        return false;
-                }
-            }
-            catch (Exception)
-            {
-                return false;
-            }
+        }
+
+        private string UpdateUrl()
+        {
+            if (HttpContext.Current == null || !Host.CheckUpgrade || !this.UserInfo.IsSuperUser)
+            {
+                return string.Empty;
+            }
+
+            return CBO.GetCachedObject<string>(new CacheItemArgs("DnnUpdateUrl"), this.RetrieveUpdateUrl);
+        }
+
+        private string RetrieveUpdateUrl(CacheItemArgs args)
+        {
+            try
+            {
+                var latestReleases = Globals.GetJsonObject<List<DTO.GithubLatestReleaseDTO>>("https://api.github.com/repos/dnnsoftware/dnn.platform/releases?per_page=5");

```

Comment:
Sould the url come from a config file maybe ?

Reply:
Although I am in favor of it, the truth is that people will never configure this. I.e. if we move to a different location, a new release and upgrade would hold the new values, but the older ones never will. So IMHO it is trying to solve something which we both see as a potential issue, but doesn't really solve it.

Reply:
Are there concerns about rate limiting if we're accessing the API without a token?